tst/EnergyPlus/unit/SurfaceGeometry.unit.cc

```diff
@@ -7021,3 +7021,532 @@ TEST_F(EnergyPlusFixture, GetSurfaceData_SurfaceOrder)
+        "WindExposed,             !- Wind Exposure",
+        "autocalculate,           !- View Factor to Ground",
+        "4,                       !- Number of Vertices",
+        "2, 4, 4,                         !- X,Y,Z  1 {m}",
+        "0, 4, 2.5,                          !- X,Y,Z  2 {m}",
+        "0, 0, 2.5,                          !- X,Y,Z  3 {m}",
+        "2, 0, 4;                         !- X,Y,Z  4 {m}",
+
+        "FenestrationSurface:Detailed,",
+        "    Skylight_C,              !- Name",
+        "    WINDOW,                  !- Surface Type",
+        "    Dbl Clr 3mm/13mm Air,    !- Construction Name",
+        "    05Roof_C,                !- Building Surface Name",
+        "    ,                        !- Outside Boundary Condition Object",
+        "    autocalculate,           !- View Factor to Ground",
+        "    ,                        !- Frame and Divider Name",
+        "    1,                       !- Multiplier",
+        "    4,                       !- Number of Vertices",
+        "    2.2, 0.2, 4,                         !- X,Y,Z  1 {m}",
+        "    5.8, 0.2, 4,                         !- X,Y,Z  2 {m}",
+        "    5.8, 3.8, 4,                         !- X,Y,Z  3 {m}",
+        "    2.2, 3.8, 4;                         !- X,Y,Z  4 {m}",
+
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    GetProjectControlData(*state, ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+
+    GetMaterialData(*state, ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+
+    GetConstructData(*state, ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+
+    GetZoneData(*state, ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+
+    SetupZoneGeometry(*state, ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+
+    GetSurfaceData(*state, ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+
+    // Center
+    EXPECT_NEAR(state->dataSurface->Surface(6).GrossArea, 16.0, 1e-6);
+    EXPECT_NEAR(state->dataSurface->Surface(6).Area, 3.04, 1e-6);
+
+    // East
+    EXPECT_NEAR(state->dataSurface->Surface(7).GrossArea, 10.0, 1e-6);
+    EXPECT_NEAR(state->dataSurface->Surface(7).Area, 10.0, 1e-6);
+
+    // West
+    EXPECT_NEAR(state->dataSurface->Surface(8).GrossArea, 10.0, 1e-6);
+    EXPECT_NEAR(state->dataSurface->Surface(8).Area, 10.0, 1e-6);
+
+    Real64 totalGrossCeilingArea = 0.0;
+    totalGrossCeilingArea =
+        state->dataSurface->Surface(6).GrossArea + state->dataSurface->Surface(7).GrossArea + state->dataSurface->Surface(8).GrossArea;
+
+    Real64 totalNetCeilingArea = 0.0;
+    totalNetCeilingArea = state->dataSurface->Surface(6).Area + state->dataSurface->Surface(7).Area + state->dataSurface->Surface(8).Area;
+
+    Real64 ceilingHeight_expected = 0.0;
+    ceilingHeight_expected = 3.25 * (state->dataSurface->Surface(7).GrossArea + state->dataSurface->Surface(8).GrossArea) / totalGrossCeilingArea +
+                             4.0 * state->dataSurface->Surface(6).GrossArea / totalGrossCeilingArea;
```

Comment:
What does the "3.25" and "4.0" signify here?

Reply:
3.25 and 4.0 (in meters) are the average height of the respective roof surfaces. This is an example of multiple roof surfaces in the same zone, with different heights--two ceiling surfaces has height of 3.25 and one has a height of 4m in the unit test case.